### PR TITLE
Cleanup trigger logging

### PR DESCRIPTION
**Pull request by Charles-Gagnon**
Part 2 of logging cleanup, same stuff as https://github.com/Azure/azure-functions-sql-extension/pull/802

With this we now have : 

13 Debug log messages - Extra information that could be useful for debugging, mostly startup/config info
2 Information messages - Telemetry opt-out and connection re-established messages
6 Warning messages - Non-critical errors
28 Error messages - Critical errors that indicate something is wrong

I plan on doing another pass after this is in to verify that the levels we have stuff at makes sense, but this should be pretty much complete at this point. Normal execution should only have a dozen or so messages per instance on startup with the occasional status update message.

### CODE REVIEW

**Comment by Charles-Gagnon** [comment]
Ah right, yeah I'll have to fix up the tests. Putting it in draft while I work on those in case I have to refactor anything...

File: samples/samples-csharp/host.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
             }
         },
         "logLevel": {
-          "default": "Information"
+          "default": "Debug"
```
Comment (Charles-Gagnon) [inline samples/samples-csharp/host.json:11]: Now that we use debug logging for more stuff I had to change this for tests. We originally had it as Information so customers didn't copy the debug settings, but in this case I think it's fine to have that vs. adding some functionality to switch the log level at runtime (which I had looked into earlier and didn't see an obvious way to do)

**Comment by Charles-Gagnon** [comment]
Ok, test fixes are in. One failing test is from the logging issue we're currently working on so unrelated to these changes.

**Review by lucyzhang929 — APPROVED** [review]

File: src/TriggerBinding/SqlTableChangeMonitor.cs [hunk 2]
```diff
@@ -717,9 +685,7 @@ private long RecomputeLastSyncVersion()
             // the only version number in the set.
             // Also this LastSyncVersion is actually updated in the GlobalState table only after verifying that the changes with
```
Comment (VasuBhog) [inline src/TriggerBinding/SqlTableChangeMonitor.cs:686]: Remove / Change comment since you removed the lastSyncVersion var.

**Review by VasuBhog — APPROVED** [review]

**Pull request merged**